src/Build/BackEnd/Components/SdkResolution/MainNodeSdkResolverService.cs

```diff
@@ -222,21 +222,33 @@ private void ProcessRequests()
-                    SdkReference sdkReference = new SdkReference(request.Name, request.Version, request.MinimumVersion);
-
-                    ILoggingService loggingService = Host.GetComponent(BuildComponentType.LoggingService) as ILoggingService;
+                    SdkResolverResponse response = null;
+                    try
+                    {
+                        // Create an SdkReference from the request
+                        SdkReference sdkReference = new SdkReference(request.Name, request.Version, request.MinimumVersion);
 
-                    // This call is usually cached so is very fast but can take longer for a new SDK that is downloaded.  Other queued threads for different SDKs will complete sooner and continue on which unblocks evaluations
-                    SdkResult result = GetSdkResultAndCache(request.SubmissionId, sdkReference, new EvaluationLoggingContext(loggingService, request.BuildEventContext, request.ProjectPath), request.ElementLocation, request.SolutionPath, request.ProjectPath);
+                        ILoggingService loggingService = Host.GetComponent(BuildComponentType.LoggingService) as ILoggingService;
 
-                    // Create a response
-                    SdkResolverResponse response = new SdkResolverResponse(result.Path, result.Version);
+                        // This call is usually cached so is very fast but can take longer for a new SDK that is downloaded.  Other queued threads for different SDKs will complete sooner and continue on which unblocks evaluations
+                        SdkResult result = GetSdkResultAndCache(request.SubmissionId, sdkReference, new EvaluationLoggingContext(loggingService, request.BuildEventContext, request.ProjectPath), request.ElementLocation, request.SolutionPath, request.ProjectPath);
 
-                    // Get the node manager and send the response back to the node that requested the SDK
-                    INodeManager nodeManager = Host.GetComponent(BuildComponentType.NodeManager) as INodeManager;
+                        // Create a response
+                        response = new SdkResolverResponse(result?.Path, result?.Version);
+                    }
+                    catch (Exception e)
+                    {
+                        // Errors cannot be logged in this background thread so it must be sent to the awaiting
+                        // node to be logged.
+                        response = new SdkResolverResponse(e);
+                    }
+                    finally
+                    {
+                        // Get the node manager and send the response back to the node that requested the SDK
+                        INodeManager nodeManager = Host.GetComponent(BuildComponentType.NodeManager) as INodeManager;
 
-                    nodeManager.SendData(request.NodeId, response);
+                        nodeManager.SendData(request.NodeId, response);
```

Comment:
So if sdk resolution errors, a packet is sent to the node, the node logs an error, which travels back as a packet to the BuildManager, 
 then the BuildManager sends a shutdown packet back to the node. I wonder if there's enough API head room for the BuildManager to shortcuit this and just fail the build directly. On the other hand I don't think this sdk failure communication happens too often, so probably it's not worth optimizing.